### PR TITLE
Remove Dynamic Loop Unroll

### PR DESCRIPTION
**Pull request by tcorringham**
The LLPC dynamic loop unroll functionality is no longer required now that
the loop unrolling in the LLVM back-end is better tuned.
As this was only used in rare cases and increases compile time when used
this change removes it completely.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit ca45d8e31a3c8100ac7ee6bf7392007d5dab80b8
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 3e28ea331cf8e7b0e5733e5bd7f7d16582c603c5</li>
<li>SPVGEN: dev 2f31d1170e8a12a66168b23235638c4bbc43ecdc</li>
<li>XGL: dev 3252b6223947f9fc67399e0798b1062983925fce</li>
<li>LLVM: amd-vulkan-dev dcb42575c3d374467bb6e4e609f8bf6d1302c297</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 927/927 (100.0%)</li>
<li>Failed: 0/927 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 68331/68331 (100.0%)</li>
<li>Failed: 0/68331 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 68363/68363 (100.0%)</li>
<li>Failed: 0/68363 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by trenouf** [comment]
Hi Tim. Could you rebase please?

**Comment by amdvlk-admin** [comment]
### Test summary for commit cc5f0ad218c823210be01c8ba73bb8d08591bb6b
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 3e28ea331cf8e7b0e5733e5bd7f7d16582c603c5</li>
<li>SPVGEN: dev 2f31d1170e8a12a66168b23235638c4bbc43ecdc</li>
<li>XGL: dev 3252b6223947f9fc67399e0798b1062983925fce</li>
<li>LLVM: amd-vulkan-dev dcb42575c3d374467bb6e4e609f8bf6d1302c297</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 927/927 (100.0%)</li>
<li>Failed: 0/927 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 68331/68331 (100.0%)</li>
<li>Failed: 0/68331 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 68363/68363 (100.0%)</li>
<li>Failed: 0/68363 (0.0%)</li>
</ul></details>


</ul></details>

File: context/llpcCompiler.cpp [hunk 1]
```diff
@@ -128,10 +127,7 @@ opt<uint32_t> ShadowDescTablePtrHigh("shadow-desc-table-ptr-high",
                                      desc("High part of VA for shadow descriptor table pointer"),
                                      init(2));
 
-// -enable-dynamic-loop-unroll: Enable dynamic loop unroll.
-opt<bool> EnableDynamicLoopUnroll("enable-dynamic-loop-unroll", desc("Enable dynamic loop unroll"), init(false));
```
Comment (linqun) [inline context/llpcCompiler.cpp]: Please bump LLPC version, and keep the definition of this option in old version. We need keep LLPC code compatible with old xgl code. thanks!

**Pull request merged**